torch_xla/csrc/tensor_util.cpp

```diff
@@ -318,39 +356,30 @@ at::Tensor MakeTensorFromXlaLiteral(const xla::Literal& literal,
                                     at::ScalarType dest_element_type) {
   xla::PrimitiveType element_type = literal.shape().element_type();
   switch (element_type) {
-    case xla::PrimitiveType::PRED: {
-      XLA_CHECK_EQ(dest_element_type, at::ScalarType::Byte);
-      return XlaLiteralToTensor<bool, uint8_t>(literal, at::ScalarType::Byte,
-                                               xla::PrimitiveType::PRED);
-    }
-    case xla::PrimitiveType::BF16: {
-      return XlaLiteralToTensor<tensorflow::bfloat16, float>(
-          literal, at::ScalarType::Float, xla::PrimitiveType::F32);
-    }
-    case xla::PrimitiveType::F32: {
-      return XlaLiteralToTensor<float, float>(literal, at::ScalarType::Float,
-                                              xla::PrimitiveType::F32);
-    }
-    case xla::PrimitiveType::U8: {
-      return XlaLiteralToTensor<xla::uint8, uint8_t>(
-          literal, at::ScalarType::Byte, xla::PrimitiveType::U8);
-    }
-    case xla::PrimitiveType::S8: {
-      return XlaLiteralToTensor<xla::int8, int8_t>(
-          literal, at::ScalarType::Char, xla::PrimitiveType::S8);
-    }
-    case xla::PrimitiveType::S16: {
-      return XlaLiteralToTensor<xla::int16, int16_t>(
-          literal, at::ScalarType::Short, xla::PrimitiveType::S16);
-    }
-    case xla::PrimitiveType::S32: {
-      return XlaLiteralToTensor<xla::int32, int32_t>(
-          literal, at::ScalarType::Int, xla::PrimitiveType::S32);
-    }
-    case xla::PrimitiveType::S64: {
-      return XlaLiteralToTensor<xla::int64, int64_t>(
-          literal, at::ScalarType::Long, xla::PrimitiveType::S64);
-    }
+    case xla::PrimitiveType::PRED:
```

Comment:
Do the tests pass on TPU? This is significantly more general than before.

Reply:
The test does not run on device. It is a simple low-level conversion matching test.
When we map ATEN ScalarType to XLA primitive type, our APIs will hand over the proper XLA type.